jps-plugin/src/org/jetbrains/jps/incremental/scala/local/LocalServer.scala

```diff
@@ -31,15 +35,15 @@ class LocalServer extends Server {
   }
 
   private def compilerFactoryFrom(sbtData: SbtData): CompilerFactory = cachedCompilerFactory.getOrElse {
-    val factory = new CachingFactory(new CompilerFactoryImpl(sbtData), 5, 5, 5)
+    val factory = new CachingFactory(new CompilerFactoryImpl(sbtData), 10, 600, 10)
```

Comment:
This numbers are quite big but we need caches to be so big. At best this should be configurable (e.g. as extension point)